src/utils/write.ts

```diff
@@ -1,48 +1,46 @@
-import { Changes, FileChange } from "../types";
 import * as vscode from "vscode";
+import { Changes, FileChange } from "../types";
 
-export async function updateFiles(
-  editor: vscode.TextEditor,
-  selection: vscode.Selection,
-  changes: Changes
-): Promise<void> {
-  await replaceSelection(editor, selection, changes);
-  await Promise.all(changes.files.map((fileChange) => updateFile(fileChange)));
-}
+export const updateFiles = (changes: Changes) => (): Promise<void> => {
+  return new Promise<void>(async (resolve) => {
+    await Promise.all(
+      changes.files.map((fileChange) => updateFile(fileChange))
+    );
+    resolve();
+  });
+};
 
-async function replaceSelection(
+export const replaceSelection = (
   editor: vscode.TextEditor,
   selection: vscode.Selection,
   changes: Changes
-): Promise<void> {
-  try {
-    await editor.edit((editBuilder) => {
-      editBuilder.replace(selection, changes.originTemplateReplacement);
-    });
-  } catch (error) {
-    errorHandler(error, "Failed to replace selected snipped");
-  }
-}
-
-async function updateFile(fileChange: FileChange): Promise<void> {
-  try {
-    vscode.workspace.fs.writeFile(
-      getUri(fileChange.path),
-      Buffer.from(fileChange.content)
-    );
-  } catch (error) {
-    errorHandler(error, `Could not edit '${fileChange.path}`);
-  }
-}
+) => () => {
+  return new Promise<void>(async (resolve, reject) => {
```

Comment:
You can write this shorter like:
```ts
return editor.edit((editBuilder) => {
  editBuilder.replace(selection, changes.originTemplateReplacement);
});
```

Reply:
still had to implement it with a `return Promise.resolve(...)` wrapped around because the return type would be `Thenable<boolean>` otherwise. But to process it a Promise is required.